### DISCUSSION TITLE
Is Function.has_user_annotations not saved in the bndb?

### DISCUSSION THREAD

**Discussion by genonullfree** [Vector35/binaryninja-api#4447] [API] [answered]
I was writing a python script that iterates over all the functions of a binary and looked for the flag `function.has_user_annotations` (https://api.binary.ninja/binaryninja.function-module.html#binaryninja.function.Function.has_user_annotations) to do some pretty prints and things. When the file is loaded in BN and modified, this property works as expected. However when I make changes, save the bndb, exit, and then reopen the bndb, this property is always set to false.

Is this property not saved in the bndb?

If it is not, what other properties are only available in the current session that don't get saved to the bndb? The API documentation does not seem to make any distinction about a currently opened and modified database vs a saved and loaded database, unless I missed it somewhere.

Is this a bug or is this intended behavior? I believe this should be a bug but I'm new to BN scripting. I'm on v3.4.4271, build b7fd028d.

Thank you for your time!

**Comment by plafosse** [comment, marked as answer]
Yes I believe this is not saved in the database and I've run into this previously.  I'm considering this a bug.

**Reply by fuzyll** [reply]
Issue is #4497.
